src/routes/safe/components/Apps/hooks/useIframeMessageHandler.ts

```diff
@@ -104,6 +101,37 @@ const useIframeMessageHandler = (
           break
         }
 
+        case SDK_MESSAGES.RPC_CALL: {
+          const payload = messagePayload as SDKMessageToPayload['RPC_CALL']
+
+          if (
+            web3ReadOnly &&
+            web3ReadOnly.currentProvider != null &&
+            typeof web3ReadOnly.currentProvider !== 'string' &&
+            'send' in web3ReadOnly.currentProvider
+          ) {
+            web3ReadOnly.currentProvider?.send?.(
+              {
+                jsonrpc: '2.0',
+                method: payload?.call,
+                params: payload?.params,
+                id: '1',
```

Comment:
out of curiosity, why won't it generate a conflict? shouldn't we make it somehow dynamic?

Reply:
The `id` is required to be defined by json rpc specification, but our callback is already mapped to a request, so I don’t think it will cause any conflict. Let’s ask @rmeissner if he could confirm it